docs/distribution/README.md

```diff
@@ -0,0 +1,923 @@
+* Rather than add a new linking API, an artifact handler would be added to registries. Many already parse the config objects to understand which platform and architectures they support.
+
+**Cons with this approach:**
+
+* An artifact handler is required that  must parse the signature object for the linked artifact
+* New code for tracking dependency linking is required through a config object handler
+* Based on the unique artifact type:  `manifest.config.mediaType: "application/vnd.cncf.notary.config.v2+jwt"`, a role check may be done to confirm the identity has a signer role.
+
+### Linking Signatures - Option 2: Distinct Linking API
+
+Similar to the manifest or index options, the client pushes the artifact and signatures through standard oci-distribution PUT apis.
+However, no linkage is made between the signature object and the signed artifact. Rather a signature linking api is added:
+
+1. Push all artifacts to the registry:  
+   * Push `net-monitor:v1` container image: `sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c`
+   * Push **acme-rockets** signature artifact `sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b`
+1. Link **acme-rockets** signature to the `net-monitor:v1` container image
+
+```HTTP
+PUT https://localhost:6000/v2/net-monitor/manifests/sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c/signatures/sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b
+```
+
+**Pros with this approach:**
+
+* Using a unique API enables RBAC to the API, without having to parse the content to determine the role check.
+
+**Cons with this approach:**
+
+* A new linking api, unique to signatures
+* The client must make two calls to achieve a single operation of uploading a signature object, which by definition has the linking information
+* If the signature linking API fails, additional garbage collection of a signature manifest must be cleaned up
+
+### Linking Signatures - Option 3: Signature Upload API
+
+In this option the signature artifact (manifest or index) is uploaded through a new signature API.
+
+**Pros with this approach:**
+
+* The signature artifact upload and role check are coupled to a signature API
+
+**Cons with this approach:**
+
+* Signature upload, which is just another artifact type, is uploaded differently than other artifacts.
+
+### Linking Signatures - Option 4: Utilize OCI Index PUT
+
+Utilizing the OCI Index, the manifest list is used to track dependencies.
+
+**Pros with this approach:**
+
+* Existing Index parsing logic is used to track despondencies
+* No additional distribution-spec APIs are required
+
+**Cons with this approach:**
+
+* OCI Index would need to support a config object:  [Add Index Support for Artifact Type #806](https://github.com/opencontainers/image-spec/issues/)
+
+## Signature Discovery
+
+Once a signature artifact is in a registry and linked to its target artifact, how is it retrieved?
```

Comment:
With the current proposed structure of the signature object, it seems Hard to track the digest of signature object in distribution via the digest of the manifest of the artifact that is signed.

Reply:
There's an ordering of artifacts issue we're trying to address:
1. wabbit-networks can push their image to docker hub
1. after it arrives, docker hub might certify it with an a docker.io signature
1. after the image is copied to acme-rockets, and tested, a new signature is added

So, the registry can and should store this linking information. Just as we do for manifests-->layers and index-->manifests

What we're doing here is making sure we can also track the reverse. If a docker image is persisted as a manifest, and the signature is persisted as an index, we need a reverse lookup:
manifest-->index

I think the question is somewhat rooted in the lack of an indexing solution in docker/distribution or the spec. What we're saying here is this will be needed. If registries choose to persist the lookup information as files, or an instance of a db is a registry operator question.
We will need to decide how we implement this in the reference implementation of docker/distribution.